### PR TITLE
chore: deprecate LoggerOps

### LINKED ISSUE #32500: deprecate logger ops, only needed for Scala 2.12

### PR DESCRIPTION
**Pull request by sebastian-alfers**
Fixes https://github.com/akka/akka/issues/32500

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/package.scala [hunk 1]
```diff
@@ -17,15 +17,15 @@ package object scaladsl {
...
-   * }}}
+   *    * }}}
    * or
    * {{{
    *import akka.actor.typed.scaladsl._
    * }}}
    *
    * @param log the underlying [[org.slf4j.Logger]]
    */
+  @deprecated("Not needed in 2.13 and later", "2.10.0")
```
Comment (johanandren) [inline akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/package.scala:28]: Isn't deprecating the class enough (and implicitly means all methods are deprecated as well)

File: akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/package.scala [hunk 2]
```diff
@@ -17,15 +17,15 @@ package object scaladsl {
    * Enable these extension methods with:
    *
    * {{{
-   * import akka.actor.typed.scaladsl.LoggerOps
-   * }}}
+   *    * }}}
```
Comment (johanandren) [inline akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/package.scala:20]: unintended change?

**Review by johanandren — COMMENTED** [review]
A few usages in the samples as well (akka-sample-sharding-scala)

File: akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/package.scala (hunk 1 above)
Comment (sebastian-alfers) [reply akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/package.scala:28]: I was not sure. If you suggest, I can use that single-deprecation approach.

**Comment by sebastian-alfers** [comment]
> A few usages in the samples as well (akka-sample-sharding-scala)

I looked again, but did not find 🙈 Can you point out a sample?

File: akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/package.scala (hunk 1 above)
Comment (johanandren) [reply akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/package.scala:29]: Just the class is enough

File: samples/akka-sample-sharding-scala/killrweather-fog/src/main/scala/sample/killrweather/fog/WeatherStation.scala [hunk 3]
```diff
@@ -57,11 +56,11 @@ private class WeatherStation(context: ActorContext[WeatherStation.Command], wsid
   private val stationUrl = s"http://${settings.host}:${httpPort}/weather/$wsid"
 
   def running: Behavior[WeatherStation.Command] = {
-    context.log.infoN(s"Started WeatherStation {} of total {} with weather port {}",
```
Comment (sebastian-alfers) [inline samples/akka-sample-sharding-scala/killrweather-fog/src/main/scala/sample/killrweather/fog/WeatherStation.scala:60]: @johanandren Isn't this what you refer to?

Comment (johanandren) [reply samples/akka-sample-sharding-scala/killrweather-fog/src/main/scala/sample/killrweather/fog/WeatherStation.scala:60]: Ah indeed, sorry, everything is good then 👍

**Review by johanandren — APPROVED** [review]
LGTM

**Pull request merged**